app/services/account_reset/delete_account.rb

```diff
@@ -0,0 +1,41 @@
+module AccountReset
+  class DeleteAccount
+    include ActiveModel::Model
+    include GrantedTokenValidator
+
+    def initialize(token)
+      @token = token
+    end
+
+    def call
+      @success = valid?
```

Comment:
I like how this is all encapsulated in the service object.  However, a mistake right here could be extremely catastrophic.  Since there are a few layers of indirection it seems too easy to make a mistake.  ie user getting set elsewhere, etc.  I'm not sure if a comment would help either.  Before it was one line before the delete: @account_reset_request.user and it was clear where the user was coming from.  If it somehow gets lost here we have armageddon.  Your thoughts?

Reply:
The concern being an endpoint with no sign in, no 2fa, and a delete.

Reply:
I think we'll be fine. We have specs to make sure the feature is working as expected. The concern about being able to finalize the account deletion without being signed in is not new to this PR. If we think we should require users to sign in with email and password before finalizing account deletion, that's a new issue we can track.

Reply:
Sounds good.